google/cloud/spanner/numeric.h

```diff
@@ -214,11 +214,11 @@ StatusOr<T> ToInteger(Numeric const& n, int exponent = 0) {
   constexpr auto kMax = (std::numeric_limits<T>::max)();
   enum { kIntPart, kFracPart } state = kIntPart;
   for (auto const ch : rep) {
-    auto dp = std::strchr(kDigits, ch);
+    auto const* dp = std::strchr(kDigits, ch);
     if (state == kFracPart) {
       if (dp - kDigits >= 5) {  // dp != nullptr
         if (v == kMax) return internal::DataLoss(rep);
-        v += 1;
+        v = static_cast<T>(v + 1);
```

Comment:
Observation: The extra annoying thing about these ones is that the previous statement always ensures that the result of the rhs expression is indeed representable in the type of the lhs.

Reply:
Ack.

Reply:
No change necessary, but for future reference, would `v += T{1}` work, or would `-Wconversion` complain about that too?

Reply:
I just tested again and `v += T{1}` and `v += static_cast<T>(1)` both produce warnings:

```
../google/cloud/spanner/numeric_test.cc:389:3:   required from here
../google/cloud/spanner/numeric.h:221:11: error: conversion from ‘int’ to ‘unsigned char’ may change value [-Werror=conversion]
  221 |         v += T{1}; // static_cast<T>(v + 1);
      |         ~~^~~~~~~
```